boot/bootutil/src/bootutil_misc.c

```diff
@@ -156,13 +161,12 @@ boot_magic_off(const struct flash_area *fap)
 int
 boot_status_entries(const struct flash_area *fap)
 {
-    switch (fap->fa_id) {
-    case FLASH_AREA_IMAGE_0:
-    case FLASH_AREA_IMAGE_1:
-        return BOOT_STATUS_STATE_COUNT * BOOT_STATUS_MAX_ENTRIES;
-    case FLASH_AREA_IMAGE_SCRATCH:
+    if (fap->fa_id == FLASH_AREA_IMAGE_SCRATCH) {
         return BOOT_STATUS_STATE_COUNT;
-    default:
+    } else if ((fap->fa_id == FLASH_AREA_IMAGE_PRIMARY) ||
```

Comment:
There are two changes being made here. One is the terminology change, the other is a stylistic change from a switch to if statements. I would argue this patch should leave this as a switch statement, and feel free to propose a separate patch that changes the switch statement into if statements (in this particular case, the if statement probably generates more code than the switch, just because of the pointer indirection, if you put the fa_id into a temporary, the code would probably be the same).

Reply:
There are five separate commits in this pull request. The terminology change and this "from switch to if" change are in two different commits.

The last change is included in the **Boot: Extend flash layout for multiple images** commit and the main reason for this:
"The FLASH_AREA_IMAGE_* object-like macros are replaced with
function-like ones and therefore some functions have been updated,
because the labels of a switch statement have to be constant
expressions."

Reply:
Ok, this makes sense, I wonder if that should be clearer from the commit text. In fact, I'd be even happier with a separate commit, before the one that adds the multiple images, that just changes the switch statements to if-blocks, with an explanation as to why that will be needed.

Reply:
This commit
https://github.com/JuulLabs-OSS/mcuboot/pull/369/commits/1547aed7dbb37b618507aaa956d45e43ec90180aí
is before the one that adds multi-image support. 
It only adds the BOOT_IMAGE_NUMBER macro and current_image variable to MCUBoot and the rest of the change is this "from switch to if" modification. It is the introduction of the current_image variable which creates the need for this and therefore, these together form a coherent commit.